homeassistant/util/package.py

```diff
@@ -28,8 +30,13 @@ def is_virtual_env() -> bool:
 
 @cache
 def is_docker_env() -> bool:
-    """Return True if we run in a docker env."""
-    return Path("/.dockerenv").exists()
+    """Return True if we run in a container env."""
+    return (
+        Path("/run/.containerenv").exists()
+        or Path("/.dockerenv").exists()
+        or os.environ.get("container")  # noqa: SIM112
+        or is_official_image()
```

Comment:
This is not a consideration here.

```suggestion
```

Reply:
I think it is - the official HA docker images always have this file, so its presence tells us we're definitely running in a container.

Reply:
It isn't relevant here. The check is only relevant is we are actually in an env.

Reply:
`is_official_image()`  is documented as "Return True if Home Assistant is running in an official container." https://github.com/home-assistant/core/blob/85ed1d2ac826fe0be41c20c0f88186d97e8adc5e/homeassistant/helpers/system_info.py#L28

That means we're running a container, which is what `is_docker_env` is determining.

Reply:
What about `is_official_image()` indicates that it's only meant to be called once we already know if we're in a container?

I don't see anything in it's implementation or documentation to indicate that intended restriction.

Reply:
I've answered already above. Please revert this change.

Reply:
> I've answered already above

I don't see your answer to "What about `is_official_image()` indicates that it's only meant to be called once we already know if we're in a container?" above. Can you please link me to it? I apologize in advance for my density.

I've reverted this change as directed, but I would love to know why, for I still think the suggestion provided by @javydekoning at https://github.com/home-assistant/core/pull/132404#issuecomment-2528363240 makes sense.